src/libsync/discovery.cpp

```diff
@@ -41,16 +41,39 @@ Q_LOGGING_CATEGORY(lcDisco, "sync.discovery", QtInfoMsg)
 bool ProcessDirectoryJob::checkForInvalidFileName(const PathTuple &path,
     const std::map<QString, Entries> &entries, Entries &entry)
 {
-    const auto originalFileName = entry.localEntry.isValid() ? entry.localEntry.name : entry.serverEntry.name;
+    if (!entry.localEntry.isValid()) {
+        return true;
+    }
+
+    const auto originalFileName = entry.localEntry.name;
     const auto newFileName = originalFileName.trimmed();
 
     if (originalFileName == newFileName) {
         return true;
     }
 
+    QString errorMessage;
+
+    if (entry.serverEntry.isValid()) {
+        errorMessage = tr("File contains trailing spaces and could not be renamed, because it is already on the server.");
+
+        auto item = SyncFileItemPtr::create();
+        if ((entry.localEntry.isValid() && entry.localEntry.isDirectory) || (entry.serverEntry.isValid() && entry.serverEntry.isDirectory)) {
+            item->_type = CSyncEnums::ItemTypeDirectory;
+        } else {
+            item->_type = CSyncEnums::ItemTypeFile;
+        }
+        item->_file = path._target;
+        item->_originalFile = path._target;
+        item->_instruction = CSYNC_INSTRUCTION_ERROR;
+        item->_status = SyncFileItem::NormalError;
+        item->_errorString = errorMessage;
+        processFileFinalize(item, path, false, ParentNotChanged, ParentNotChanged);
```

Comment:
looks good but can you add a test case for it ?
looks like we still will get the proper debug output so indeed your approach is working